sql/sql_class.cc

```diff
@@ -895,13 +896,44 @@ void THD::cleanup_connection(void) {
 #endif
 }
 
-void THD::set_connection_certificate(std::string const &cert) {
-  DBUG_ASSERT(m_connection_certificate.empty());
-  m_connection_certificate = cert;
-}
+std::string THD::extract_peer_certificate_info(const THD *thd, bool printable) {
+  if (!thd->has_net_vio_ssl_arg()) return {};
+
+  auto ssl = static_cast<SSL *>(thd->get_net_vio_ssl_arg());
```

Comment:
This could be const SSL*, and the get_net_vio getters could also return const pointers.

Reply:
Good point. Reworked.